staging/src/k8s.io/apimachinery/pkg/watch/retrywatcher.go

```diff
@@ -0,0 +1,151 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package watch
+
+import (
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+// Watch function is responsible for creating a watcher starting at sinceResourceVersion.
+type WatchFunc func(sinceResourceVersion string) Interface
+
+// Type of an event that will be returned if RetryWatcher fails
+type RetryWatcherError string
+
+// Implements interface.
+func (obj RetryWatcherError) GetObjectKind() schema.ObjectKind { return schema.EmptyObjectKind }
+
+// Implements interface.
+func (obj RetryWatcherError) DeepCopyObject() runtime.Object { return obj }
+
+// Interface to get resource version from events.
+// We can't reuse an interface from meta otherwise it would be a cyclic dependency and we need just this one method
+type resourceVersionInterface interface {
```

Comment:
I'd call this `resourceVersionGetter` but feel free to ignore if you disagree :)

Reply:
I like it more too :)